Add Workspace mapping documentation (#1649)
astronomer/docs: astro/view-clusters.md

---
sidebar_label: "Configure clusters"
title: "Configure Astro clusters"
id: view-clusters
description: View information about clusters in the Cloud UI.
---

import PremiumBadge from '@site/src/components/PremiumBadge';

Use the Cloud UI to get an overview of your Organization's clusters and retrieve cluster information that may be required by Astronomer support.

> Some cluster changes can be made only by Astronomer. See [Request a cluster change](modify-cluster.md).

Reviewer: ```suggestion
Some cluster changes can be made only by Astronomer support. See [Request a cluster change](modify-cluster.md).
```

## View all clusters

> In the Cloud UI, click the **Clusters** tab to view information about the clusters in your Organization. The following table provides descriptions for each cluster value.

Reviewer: ```suggestion
In the Cloud UI, click **Clusters** to view information about the clusters in your Organization. The following table provides descriptions for each cluster value.
```

| Updated            | The date and time the cluster's settings were last updated                                                                  |
| Created            | The date and time the cluster was created                                                                          |

> All users in your Organization have access to this view. For more information about each value, see [Cluster settings reference](https://docs.astronomer.io/astro/category/cluster-settings).

Reviewer: ```suggestion
All users in your Organization have access to the **Clusters** page. For more information about each value, see [Cluster settings reference](https://docs.astronomer.io/astro/category/cluster-settings).
```

## Configure a cluster

> To view and update settings for a specific cluster, click the cluster's name in the **Clusters** tab. A cluster's configuration menu includes the following pages:

Reviewer: ```suggestion
To view and update settings for a specific cluster, click the cluster name on the **Clusters** page. A cluster's configuration menu includes the following pages:
```

> - **Worker Types**: Shows all available worker types on the cluster. To add or remove a worker type on a cluster, see [Request a cluster change](modify-cluster.md).

Reviewer: ```suggestion
- **Worker Types**: A list of all available worker types on the cluster. To add or remove a worker type on a cluster, see [Request a cluster change](modify-cluster.md).
```

> - **Workspace Authorization**: Shows which Workspaces are allowed to create Deployments on the cluster.

Reviewer: ```suggestion
- **Workspace Authorization**: A list of Workspaces with permissions to create Deployments on the cluster.
```

> - **Details**: Shows the same information available from the  **Clusters** tab. See [View all clusters](#view-all-clusters).

Reviewer: ```suggestion
- **Details**: Duplicates the information provided on the  **Clusters** tpage. See [View all clusters](#view-all-clusters).
```
This doesn't appear to be true:

![image](https://user-images.githubusercontent.com/104205257/208099275-e6659d1f-ff29-424c-acaf-db7fd5b9f505.png)

Perhaps the UI has changed?

### Limit Deployment creation on a cluster 

<PremiumBadge />

> You can configure a cluster so that it contains Deployments only from specific Workspaces that are mapped to the cluster. Use this feature to establish fine-grained control over who can use your cluster resources and how.

Reviewer: ```suggestion
You can configure a cluster so that it contains Deployments only from specific Workspaces that are mapped to the cluster. Use this feature to limit how specific users can use your cluster resources.
```
Why is this beneficial? Why should I care?

> 

Reviewer 2: These steps can only be performed by users with Organization Owner role

> 1. In the Cloud UI, go to the **Clusters** tab, select a cluster, and then go to **Workspace Authorization**.

Reviewer 2: Need to click Edit Workspace Authorization

> 2. Check each Workspace that you want to map to the cluster. 

Reviewer: ```suggestion
2. Select the Workspaces that you want to map to the cluster. 
```

Reviewer 2: You have to click `Restricted` first for the list of Workspaces to show up.

> 3. Click **Save**.

Author: @chomipi88 took a random guess at this because I didn't have the feature enabled on the cluster I belonged to

> 3. Click **Save**.

Reviewer 2: ```suggestion
3. Click **Update**.
```

> :::info 

Reviewer 2: We should also add that once a cluster is marked as Restricted, new Workspaces will need to be explicitly enabled for that cluster. If attempting to create a Deployment in a new Workspace, a Restricted cluster will not appear in the drop down until enabled.

> A cluster with mapped Workspaces can't host Deployments from any Workspaces that aren't mapped to the cluster. To map Workspaces to a cluster, you must first transfer any existing Deployments on the cluster to one of these Workspace.

Reviewer: ```suggestion
A cluster with mapped Workspaces can't host Deployments from any Workspaces that aren't mapped to the cluster. To map Workspaces to a specific cluster, you must transfer all existing Deployments on the cluster to one of these Workspaces.
```

Reviewer 2: Instead of mapped, lets use the word authorized.

> Similarly, to unmap a Workspace but keep its Deployments in the cluster, you must transfer your Deployments to a new cluster which is still mapped to the Workspace. See [Transfer a Deployment to another Workspace](configure-deployment-resources.md#transfer-a-deployment-to-another-workspace).

Reviewer: ```suggestion
To remove a mapped Workspace and keep its Deployments in the cluster, you must transfer your Deployments to a new cluster which is mapped to the Workspace. See [Transfer a Deployment to another Workspace](configure-deployment-resources.md#transfer-a-deployment-to-another-workspace).
```

Reviewer 2: The transfer would happen to another authorized Workspace. We don't have the capability to transfer deployments to other clusters.

| Updated            | The date and time the cluster's settings were last updated                                                                  |
| Created            | The date and time the cluster was created                                                                          |

All users in your Organization have access to the **Clusters** page. For more information about each value, see [Cluster settings reference](https://docs.astronomer.io/astro/category/cluster-settings).

## Configure a cluster

To view and update settings for a specific cluster, click the cluster name on the **Clusters** page. The configuration details page for the cluster includes the following tabs:

- **Worker Types**: A list of all available worker types on the cluster. To add or remove a worker type on a cluster, see [Request a cluster change](modify-cluster.md).
- **Workspace Authorization**: Shows which Workspaces that are authorized to create Deployments on the cluster.
- **Details**: A list of all available configuration information for the cluster. See [View all clusters](#view-all-clusters).

> ### Authorize Workspaces to a Cluster

Reviewer 2: @kwcanuck can we add a public preview banner similar to Cloud IDE with the below text?

`This feature is currently in [Public Preview](https://docs.astronomer.io/astro/feature-previews). If you have any feedback, submit a support ticket or contact your Customer Success Manager.`
